ACKategories-iOS/Base/FlowCoordinator.swift

```diff
@@ -80,45 +80,48 @@ extension Base {
                 animationGroup.enter()
                 $0.stop(animated: animated, completion: animationGroup.leave)
             }
-
-            // dismiss all VCs presented from root or nav
-            if rootViewController.presentedViewController != nil {
-                animationGroup.enter()
-                rootViewController.dismiss(animated: animated, completion: animationGroup.leave)
-            }
-
-            // pop all view controllers when started within navigation controller
-            if let navigationController = navigationController, let index = navigationController.viewControllers.firstIndex(of: rootViewController) {
-                // VCs to be removed from navigation stack
-                let toRemoveViewControllers = navigationController.viewControllers[index..<navigationController.viewControllers.count]
-
-                // dismiss all presented VCs on VCs to be removed
-                toRemoveViewControllers.forEach { vc in
-                    if vc.presentedViewController != nil {
-                        animationGroup.enter()
-                        vc.dismiss(animated: animated, completion: animationGroup.leave)
+            
+            if let rootViewController = rootViewController {
```

Comment:
I think it'd be nice to take this opportunity and refactor this method a little bit by dividing it into private helper(s). At least the body in following this `if let`

Reply:
I also think that this can be tested now that we have flow coordinators tests (just start up some flow, set the `rootViewController` to `nil` and run `stop()` and see what happens)

Reply:
see the newest version for those changes ☺️